neonvm-daemon/cmd/main.go

```diff
@@ -26,24 +32,71 @@ func main() {
 	defer logger.Sync() //nolint:errcheck // what are we gonna do, log something about it?
 
 	logger.Info("Starting neonvm-daemon", zap.String("addr", *addr))
+	srv := cpuServer{
+		cpuOperationsMutex: &sync.Mutex{},
+		cpuScaler:          cpuscaling.NewCPUScaler(),
+		logger:             logger.Named("cpu-srv"),
+	}
+	srv.run(*addr)
+}
 
-	srv := cpuServer{}
-	srv.run(logger, *addr)
+type cpuServer struct {
+	// Protects CPU operations from concurrent access to prevent multiple ensureOnlineCPUs calls from running concurrently
+	// and ensure that status response is always actual
+	cpuOperationsMutex *sync.Mutex
+	cpuScaler          *cpuscaling.CPUScaler
+	logger             *zap.Logger
 }
 
-type cpuServer struct{}
+func (s *cpuServer) handleGetCPUStatus(w http.ResponseWriter) {
```

Comment:
same comment re: whitespace on `handleGetCPUStatus` and `handleSetCPUStatus` — just a couple blank lines to provide visual distinction to the different parts should suffice